src/test/java/labDemoTest.java

```diff
@@ -64,73 +90,99 @@ void test_button_unchanged() throws FindFailed {
+        // Find text "Editor" and click on it
+        screen.findText("Editor").click();
 
-        for (File file : imgList != null ? imgList : new File[0]) {
+        // Create patterns for pane closed and pane open
+        Pattern firstPaneClosed = new Pattern("firstPaneClosed");
+        Pattern firstPaneOpen = new Pattern("firstPaneOpen");
 
-            if (file.getName().equals("6-text.png")) {
-                break;
-            }
+        // Determine whether the tab is closed or open
+        if (screen.exists(firstPaneClosed.similar(0.8)) != null) {
+            // If the pane was closed then click on it to open the pane
+            // Wait for Sikuli to find the place similar to the picture of closed pane and click on it to open it
+            screen.wait(firstPaneClosed.similar(0.8), 2).click();
+        }
 
-            Pattern img = new Pattern(file.getPath());
-            labDemo.getMyScreen().wait(img.similar(0.9), 2).click();
-            labDemo.getMyScreen().mouseUp();
+        // Wait for Sikuli to find the text box and click on it to activate the writing functionality
+        screen.wait(new Pattern("textField").similar(0.9), 2).click();
+        // Write text to the text box
+        screen.write("Tere! Hi! Hola!" + Key.ENTER + "Go little Rock Star!");
+        // Wait for Sikuli to find the place similar to the picture of pane named "First" and click on it to close
+        screen.wait(firstPaneOpen.similar(0.8), 2).click();
+        // Click on the same place on the screen to open the tab again
+        screen.click();
 
-            if (file.getName().equals("3-text.png")) {
-                labDemo.getMyScreen().write("Tere! Hi! Hola!" + Key.ENTER + "Go little Rock Star!");
-            }
-        }
+        // Assert if the written lines are present on the screen
+        Assertions.assertNotNull(screen.existsText("Tere! Hi! Hola!", 2));
+        Assertions.assertNotNull(screen.existsText("Go little Rock Star!", 2));
 
-        for (File file : imgList != null ? imgList : new File[0]) {
-            if (file.getName().equals("3-text.png")) {
-                Pattern text = new Pattern("6-text.png");
-                System.out.println(labDemo.getMyScreen().exists(text.exact(), 2));
-                Assertions.assertNotNull(labDemo.getMyScreen().exists(text.exact(), 2));
-                break;
-            }
-            Pattern img = new Pattern(file.getPath());
-            labDemo.getMyScreen().wait(img, 2).click();
-        }
     }
 
+
+    /**
+     * Tab 3 (Copyable):
+     *
+     * General:
+     *   * The tab consists of 5 words.
+     *   * Colors of the words are generated randomly every time you run the app.
+     * Functionality to test:
+     *   * Each word should be copyable. [“test_copyable”]
+     *
+     * This test should fail, because two words are not copiable.
+     */
+
     @Test
     void test_copyable() throws InterruptedException, FindFailed {
```

Comment:
This test does not really do what it is supposed to. I tried to rewrite it according to the previous solution:

```
def test_copiable(self):
        time.sleep(1)
        words = self.h.reg.collectWords()    
        check = ["First", "Second", "Third", "Forth", "Fifth"]
        for i in range(len(words)):
            doubleClick(words[i])
            self.assertEqual(copySelectedText(), check[i])
```

but I could not find a way to gracefully find the region from where to find the words. Without the region, `findAllWords` finds all the words from all over the screen and not even including the needed five. When I tried to find the words with a for loop over the list of words like: `findWord("First")`, it could not find the word "Second". I guess it has something to do with the fact that this word is also not copiable, although the `collectWords()` seemed to work in the previous solution.

This test is very difficult to think through, at least for me.